src/mlpack/methods/ann/layer/convolution.hpp

```diff
@@ -314,11 +317,11 @@ class ConvolutionType : public Layer<MatType>
    * @param input The input data to be rotated.
    * @param output The rotated output.
    */
-  template<typename eT>
-  void Rotate180(const arma::Mat<eT>& input, arma::Mat<eT>& output)
+  void Rotate180(const MatType& input, MatType& output,
+      const typename std::enable_if_t<IsMatrix<MatType>::value>* = 0)
```

Comment:
If `Rotate180()` is only used internal to this class, and we're already checking that things have the right types at the user-facing functions, then we can drop the SFINAE check here and just leave it as documentation.  (I might suggest that just because every SFINAE check makes compilation take just a tiny bit longer.)

Reply:
Well, will see if the compilation passes with or without